product-service/services/product.service.test.ts

```diff
@@ -0,0 +1,41 @@
+import 'jest';
+import { describe, test, expect, beforeEach } from '@jest/globals';
+import { ProductService } from './product-service';
+import products from './productList.json';
+
+describe('ProductService', () => {
+    let productService: ProductService;
+    beforeEach(() => {
+        productService = new ProductService();
+    });
+
+    test('Should return products array on getProducts', async () => {
+        const data = await productService.getProducts();
+        expect(data).toEqual(products);
+    })
+
+    test('Should return error object for wrong productid from getProdutById', async () => {
+        try {
+            await productService.getProductById('123');
+        } catch(err) {
+            expect(err).toEqual({
+                message: 'Product Not found',
+                errorCode: 404
+            });
+        }
+    })
+
+    test('Should return correct data for productid from getProdutById', async () => {
```

Comment:
typo getProdutById

Reply:
Updated in entire file. Thanks for pointing!